src/test/kotlin/nomic/unit/config/security/filters/BasicAuthenticationFilterTest.kt

```diff
@@ -0,0 +1,181 @@
+
+        mockMvc.perform(request)
+            .andExpect(status().isOk)
+            .andExpect(content().string("null"))
+    }
+
+    @Test
+    fun filter_validEndpoint_badLogin_noHeader_fails() {
+        val request = MockMvcRequestBuilders.get("/api/auth/token")
+            .header(HttpHeaders.AUTHORIZATION, "${encode(invalidLogin)}")
+
+        mockMvc.perform(request)
+            .andExpect(status().isOk)
+            .andExpect(content().string("null"))
+    }
+
+    @Test
+    fun filter_validEndpoint_badLoginName_fails() {
+        val request = MockMvcRequestBuilders.get("/api/auth/token")
+            .header(HttpHeaders.AUTHORIZATION, "${encode("To be or not to be", "ThatIsTheQuestion")}")
+
+        mockMvc.perform(request)
+            .andExpect(status().isOk)
+            .andExpect(content().string("null"))
+    }
+
+    @Test
+    fun filter_invalidEndpoint_noLogin_fails() {
+        mockMvc.perform(MockMvcRequestBuilders.get("/test"))
+            .andExpect(content().string("null"))
+            .andExpect(status().isOk)
+    }
+
+    @Test
+    fun filter_invalidEndpoint_validLogin_fails() {
+        val request = MockMvcRequestBuilders.get("/test")
+            .header(HttpHeaders.AUTHORIZATION, "Basic ${encode(login1)}")
+
+        mockMvc.perform(request)
+            .andExpect(content().string("null"))
+            .andExpect(status().isOk)
+    }
+
+    @Test
+    fun filter_invalidEndpoint_invalidLogin_fails() {
+        val request = MockMvcRequestBuilders.get("/test")
+            .header(HttpHeaders.AUTHORIZATION, "Basic ${encode(invalidLogin)}")
+
+        mockMvc.perform(request)
+            .andExpect(content().string("null"))
+            .andExpect(status().isOk)
+    }
+
+    @Test
+    fun filter_invalidEndpoint_badLogin_noHeader_fails() {
+        val request = MockMvcRequestBuilders.get("/test")
+            .header(HttpHeaders.AUTHORIZATION, "${encode(invalidLogin)}")
+
+        mockMvc.perform(request)
+            .andExpect(status().isOk)
+            .andExpect(content().string("null"))
+    }
+
+    @Test
+    fun filter_invalidEndpoint_badLoginName_fails() {
+        val request = MockMvcRequestBuilders.get("/test")
+            .header(HttpHeaders.AUTHORIZATION, "${encode("To be or not to be", "ThatIsTheQuestion")}")
+
+        mockMvc.perform(request)
+            .andExpect(status().isOk)
+            .andExpect(content().string("null"))
+    }
+
+    @RestController
+    private class TestController {
+
+        private fun core(): String {
+            val context = SecurityContextHolder.getContext()
+            val user = context.authentication?.principal as User?
+            return user?.id.toString()
+        }
+
+        @GetMapping("/api/auth/token")
+        fun testValid(): String {
+            return core()
+        }
+
+        @GetMapping("/test")
+        fun testInvalid(): String {
+            return core()
+        }
+    }
```

Comment:
There should be a comment explaining why the mock is here and what purpose it is serving to the test